central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -245,10 +279,87 @@ func (m *managerImpl) ProcessRescanRequest(_ context.Context, _ interface{}) err
 
 // DeleteScan processes a request to delete an existing compliance scan configuration.
 // TODO(ROX-19540)
-func (m *managerImpl) DeleteScan(_ context.Context, _ interface{}) error {
+func (m *managerImpl) DeleteScan(ctx context.Context, scanID string) error {
 	// TODO:
 	// 1. Validate config exists in database
 	// 2. Lock config so it cannot be edited/updated
 	// 3. Push request to Sensor
-	panic("implement me")
+
+	// Check if scan configuration already exists.
+	scanConfig, found, err := m.scanSettingDS.GetScanConfiguration(ctx, scanID)
+	if err != nil {
+		log.Error(err)
+		return errors.Wrapf(err, "Unable to delete scan configuration ID %q.", scanID)
+	}
+
+	if !found {
+		return errors.Errorf("Scan configuration ID %q does not exist.", scanID)
+	}
+
+	if scanConfig == nil {
+		return errors.Errorf("Scan configuration ID %q is nil.", scanID)
+	}
+
+	scanStatus, err := m.scanSettingDS.GetScanConfigClusterStatus(ctx, scanID)
+	if err != nil {
+		log.Error(err)
+		return errors.Wrapf(err, "Unable to delete scan configuration ID %q.", scanID)
+	}
+
+	err = m.scanSettingDS.LockScanConfiguration(ctx, scanID)
```

Comment:
This feels like a abstraction leak as it directly exposes lock functionality out of a class. What is the purpose of this given that the `UpdateClusterStatus` and `DeleteScanConfiguration` all will lock it again but with a different scan id?

Reply:
that's a good question, that's something I am not too sure about, I wonder if when we start process the request on scan deletion, how would we need to lock it and prevent other process from making changing to it.